spynnaker8/utilities/spynnaker8_neo_segment.py

```diff
@@ -50,21 +57,20 @@ def read_in_spikes(self, spikes, t, ids, indexes, first_id,
         t_stop = t * pq.ms
 
         for (id, index) in zip(ids, indexes):
+            spiketrain = SpikeTrain(
+                times=spikes[spikes[:, 0] == id - first_id][:, 1],
+                t_start=recording_start_time,
+                t_stop=t_stop,
+                units='ms',
+                source_population=label,
+                source_id=id,
+                source_index=index)
             # get times per atom
-            self.spiketrains.append(
-                SpikeTrain(
-                    times=spikes[spikes[:, 0] ==
-                                 id - first_id][:, 1],
-                    t_start=recording_start_time,
-                    t_stop=t_stop,
-                    units='ms',
-                    source_population=label,
-                    source_id=id,
-                    source_index=index))
+            self.spiketrains.append(spiketrain)
 
     @staticmethod
     def _convert_extracted_data_into_neo_expected_format(
-            signal_array, channel_indices):
+            signal_array, indexes):
```

Comment:
Fun! The use of “indices” vs “indexes” is tricky indeed, but since the usage here is mainly like that in mathematical situations, I'd recommend using “indices”. I do like the dropping of the prefix though.

For a reasonable discussion, see: http://www.worldwidewords.org/qa/qa-ind2.htm

Reply:
Note: This class is removed in https://github.com/SpiNNakerManchester/sPyNNaker8/pull/69

The removal of the prefix is critical as in pynn 0.9 channel_indices has changed to a different Object. See: https://github.com/NeuralEnsemble/python-neo/blob/master/neo/core/channelindex.py

neo uses ".index" to hold the data so if anything we should also use "index"